### PR TITLE
lightning-loop: init at version 0.12.2

### PR DESCRIPTION
**Pull request by ProofOfKeags**
###### Motivation for this change

This adds the Lightning Loop Client to nixpkgs

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/blockchains/lightning-loop.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchFromGitHub {
+    owner = "lightninglabs";
+    repo = "loop";
+    rev = "v${version}";
+    sha256 = "1yyx9j9lxdianm0mxm8zs2yn6xglc1kb7vbcalnxss8j62zn1msm";
+  };
+
+  vendorSha256 = "03z0cmn9qgcmqm8llybfn1hz1m9hx3pn18m11s3fwnay8ib00r89";
+
+  subPackages = ["cmd/loop" "cmd/loopd"];
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/lightning-loop.nix:19]:
```suggestion
  subPackages = [ "cmd/loop" "cmd/loopd" ];
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -27541,6 +27541,8 @@ in
 
   ledger-live-desktop = callPackage ../applications/blockchains/ledger-live-desktop { };
 
+  lightning-loop = callPackage ../applications/blockchains/lightning-loop.nix { };
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:27544]: Packages shouldn't be packaged like this and go into their own directory with default.nix.

[the remark this answers is not in the document]
Comment (ProofOfKeags) [reply pkgs/top-level/all-packages.nix:27544]: fixed by bf818ac

[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:27544]:
```suggestion
  lightning-loop = callPackage ../applications/blockchains/lightning-loop { };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by ProofOfKeags** [comment]
Cool, sorry for that.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
please squash all the package related commits together so that we have two commits in the end: one for the maintainer and one for the package.

**Pull request merged**